cmd/sourced/compose/workdir/common.go

```diff
@@ -191,26 +191,38 @@ func ListPaths() ([]string, error) {
 	return res, nil
 }
 
-// RemovePath removes working directory by removing required files
+// RemovePath removes working directory by removing its files,
 // and recursively removes directories up to the workdirs root as long as they are empty
 func RemovePath(path string) error {
 	workdirsRoot, err := workdirsPath()
 	if err != nil {
 		return err
 	}
 
-	for _, f := range RequiredFiles {
-		if err := os.Remove(filepath.Join(path, f)); err != nil {
-			return errors.Wrap(err, "could not remove from workdir directory")
-		}
-	}
-
 	for {
-		files, err := ioutil.ReadDir(path)
+		containsDirs := false
+		err := filepath.Walk(path, func(item string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if info.IsDir() {
+				containsDirs = true
+				return nil
+			}
+
+			if err := os.Remove(item); err != nil {
+				return errors.Wrap(err, "could not remove from workdir directory")
+			}
+
+			return nil
+		})
+
 		if err != nil {
-			return errors.Wrap(err, "could not read workdir directory")
+			return err
 		}
-		if len(files) > 0 {
+
+		if !containsDirs {
```

Comment:
`os.Remove` will fail below because the code will be executed only when the path directory isn't empty